pex/venv/virtualenv.py

```diff
@@ -84,12 +84,19 @@ def find_site_packages_dir(
     interpreter=None,  # type: Optional[PythonInterpreter]
 ):
     # type: (...) -> str
+
+    real_venv_dir = os.path.realpath(venv_dir)
+    site_packages_dirs = OrderedSet()  # type: OrderedSet[str]
+
     interpreter = interpreter or PythonInterpreter.get()
-    site_packages_dirs = []
-    for path in OrderedSet(os.path.realpath(entry) for entry in interpreter.sys_path):
-        _, tail = os.path.split(path)
-        if "site-packages" == tail and os.path.isdir(path):
-            site_packages_dirs.append(path)
+    for entry in interpreter.sys_path:
+        real_entry_path = os.path.realpath(entry)
+        if os.path.commonprefix((real_venv_dir, real_entry_path)) != real_venv_dir:
+            # This ignores system site packages when the venv is built with --system-site-packages.
```

Comment:
nice! Alternatively `if not real_entry_path.startswith(real_venv_dir)` is a bit more readable, but maybe has some edge cases that the os.path modules would handle.

Reply:
Well, you hit something that has always bothered me. What you suggested is not good since `a/bad/match` starts with `a/b` but the real common prefix is `a`. That said - what I'm using does this same badness!:
```
jsirois@Gill-Windows:~/dev/pantsbuild/jsirois-pex (main) $ python2.7 -c 'import os; print(os.path.commonprefix(["a/b", "a/bad/match"]))'
a/b
jsirois@Gill-Windows:~/dev/pantsbuild/jsirois-pex (main) $ python3.11 -c 'import os; print(os.path.commonprefix(["a/b", "a/bad/match"]))'
a/b
```

I just sort of obstinately use it since some from `os.path` should really do the right thing. What I really should do is write a utility and re-use that throughout Pex: https://github.com/pantsbuild/pex/issues/1916

Reply:
It looks like the fix came in 3.5: https://docs.python.org/3/library/os.path.html#os.path.commonpath